shap/explainers/tree.py

```diff
@@ -366,6 +385,150 @@ def recurse(node):
+
+    def create_tree(self, nodes_dict, scale):
+        """ Creates a tree given a dictionary representation of a tree.
+
+        Parameters
+        -------
+        nodes_dict : A dictionary that contains a single tree.  For example:
+        {0: '[f1<0] yes=1,no=2,missing=1,gain=4500,cover=2000',
+         1: '[f0<0] yes=3,no=4,missing=3,gain=1000,cover=1000',
+         2: '[f0<0] yes=5,no=6,missing=5,gain=1000,cover=1000',
+         3: 'leaf=0.5,cover=500',
+         4: 'leaf=2.5,cover=500',
+         5: 'leaf=-2.5,cover=500',
+         6: 'leaf=-0.5,cover=500'}
+
+        Returns
+        -------
+        A Tree object.
+        """
+        m = max(nodes_dict.keys())+1
+        children_left = -1*np.ones(m,dtype="int32")
+        children_right = -1*np.ones(m,dtype="int32")
+        children_default = -1*np.ones(m,dtype="int32")
+        features = -2*np.ones(m,dtype="int32")
+        thresholds = -1*np.ones(m,dtype="float64")
+        values = 1*np.ones(m,dtype="float64")
+        node_sample_weights = np.zeros(m,dtype="float64")
+        values_lst = list(nodes_dict.values())
+        keys_lst = list(nodes_dict.keys())
+        for i in range(0,len(keys_lst)):
+            value = values_lst[i]
+            key = keys_lst[i]
+            if ("leaf" in value):
+                # Extract values
+                val = float(value.split("leaf=")[1].split(",")[0])
+                node_sample_weight = float(value.split("cover=")[1])
+                # Append to lists
+                values[key] = val
+                node_sample_weights[key] = node_sample_weight
+            else:
+                c_left = int(value.split("yes=")[1].split(",")[0])
+                c_right = int(value.split("no=")[1].split(",")[0])
+                c_default = int(value.split("missing=")[1].split(",")[0])
+                feat_thres = value.split(" ")[0]
+                if ("<" in feat_thres):
+                    feature = int(feat_thres.split("<")[0][2:])
+                    threshold = float(feat_thres.split("<")[1][:-1])
+                if ("=" in feat_thres):
+                    feature = int(feat_thres.split("=")[0][2:])
+                    threshold = float(feat_thres.split("=")[1][:-1])
+                node_sample_weight = float(value.split("cover=")[1].split(",")[0])
+                children_left[key] = c_left
+                children_right[key] = c_right
+                children_default[key] = c_default
+                features[key] = feature
+                thresholds[key] = threshold
+                node_sample_weights[key] = node_sample_weight
+        tree_dict = {"children_left":children_left, "children_right":children_right,
+                     "children_default":children_default, "feature":features,
+                     "threshold":thresholds, "value":values[:,np.newaxis],
+                     "node_sample_weight":node_sample_weights}
+        return(Tree(tree_dict))
+
+    def independent_treeshap(self,x,y=None):
```

Comment:
we won't expect users to call this directly since with the `feature_dependence` option in the constructor the `shap_values` method will know to call this when we are assuming independence.